Compile error with canonicalize_file_name

When compiling with GCC 8.3.0 I am getting this error in the goby_moos module:

```
[ 70%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/goby_moos_app.cpp.o
In file included from /apk_src/goby/src/goby3-3.0/src/apps/middleware/gobyd/gobyd.cpp:22:
/apk_src/goby/src/goby3-3.0/build/include/goby/common/application_base3.h: In constructor 'goby::common::ApplicationBase3<Config>::ApplicationBase3()':
/apk_src/goby/src/goby3-3.0/build/include/goby/common/application_base3.h:180:30: error: there are no arguments to 'canonicalize_file_name' that depend on a template parameter, so a declaration of 'canonicalize_file_name' must be available [-fpermissive]
         int result = symlink(canonicalize_file_name(file_name.c_str()), file_symlink.c_str());
                              ^~~~~~~~~~~~~~~~~~~~~~
/apk_src/goby/src/goby3-3.0/build/include/goby/common/application_base3.h:180:30: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
/apk_src/goby/src/goby3-3.0/build/include/goby/common/application_base3.h: In instantiation of 'goby::common::ApplicationBase3<Config>::ApplicationBase3() [with Config = goby::protobuf::GobyDaemonConfig]':
/apk_src/goby/src/goby3-3.0/src/apps/middleware/gobyd/gobyd.cpp:69:63:   required from here
/apk_src/goby/src/goby3-3.0/build/include/goby/common/application_base3.h:180:52: error: 'canonicalize_file_name' was not declared in this scope
         int result = symlink(canonicalize_file_name(file_name.c_str()), file_symlink.c_str());
                              ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
[ 70%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/modem_id_convert.cpp.o
[ 71%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/moos_translator.cpp.o
make[2]: *** [src/apps/middleware/gobyd/CMakeFiles/gobyd.dir/build.make:63: src/apps/middleware/gobyd/CMakeFiles/gobyd.dir/gobyd.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:979: src/apps/middleware/gobyd/CMakeFiles/gobyd.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 71%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/moos_protobuf_helpers.cpp.o
[ 71%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/moos_ufield_sim_driver.cpp.o
[ 72%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/moos_bluefin_driver.cpp.o
[ 72%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/transitional/message_val.cpp.o
[ 73%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/transitional/message_algorithms.cpp.o
[ 73%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/moos_geodesy.cpp.o
[ 74%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/frontseat/frontseat.cpp.o
[ 74%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/middleware/moos_plugin_translator.cpp.o
In file included from /apk_src/goby/src/goby3-3.0/build/include/goby/middleware/multi-thread-application.h:28,
                 from /apk_src/goby/src/goby3-3.0/src/moos/middleware/moos_plugin_translator.h:27,
                 from /apk_src/goby/src/goby3-3.0/src/moos/middleware/moos_plugin_translator.cpp:23:
/apk_src/goby/src/goby3-3.0/build/include/goby/common/application_base3.h: In constructor 'goby::common::ApplicationBase3<Config>::ApplicationBase3()':
/apk_src/goby/src/goby3-3.0/build/include/goby/common/application_base3.h:180:30: error: there are no arguments to 'canonicalize_file_name' that depend on a template parameter, so a declaration of 'canonicalize_file_name' must be available [-fpermissive]
         int result = symlink(canonicalize_file_name(file_name.c_str()), file_symlink.c_str());
                              ^~~~~~~~~~~~~~~~~~~~~~
/apk_src/goby/src/goby3-3.0/build/include/goby/common/application_base3.h:180:30: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
[ 75%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/__/__/include/goby/moos/protobuf/bluefin_driver.pb.cc.o
[ 75%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/__/__/include/goby/moos/protobuf/ctd_sample.pb.cc.o
[ 76%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/__/__/include/goby/moos/protobuf/desired_course.pb.cc.o
[ 76%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/__/__/include/goby/moos/protobuf/frontseat.pb.cc.o
[ 77%] Building CXX object src/moos/CMakeFiles/goby_moos.dir/__/__/include/goby/moos/protobuf/frontseat_config.pb.cc.o
make[2]: *** [src/moos/CMakeFiles/goby_moos.dir/build.make:297: src/moos/CMakeFiles/goby_moos.dir/middleware/moos_plugin_translator.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:500: src/moos/CMakeFiles/goby_moos.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
>>> ERROR: goby: build failed
```

--- a/src/common/application_base3.h
+++ b/src/common/application_base3.h
@@ -177,7 +177,7 @@ template <typename Config> goby::common::ApplicationBase3<Config>::ApplicationBa
                                  << std::endl;
 
         remove(file_symlink.c_str());
-        int result = symlink(canonicalize_file_name(file_name.c_str()), file_symlink.c_str());
+        int result = symlink(realpath(file_name.c_str(), NULL), file_symlink.c_str());
         if (result != 0)
             glog.is(WARN) &&
                 glog << "Cannot create symlink to latest file. Continuing onwards anyway"

